src/rapids_pre_commit_hooks/alpha_spec.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import reduce
+
+import yaml
+from packaging.requirements import Requirement
+from packaging.specifiers import SpecifierSet
+
+from .lint import LintMain
+
+RAPIDS_ALPHA_SPEC_PACKAGES = {
+    "rmm",
+    "pylibcugraphops",
+    "pylibcugraph",
+    "nx-cugraph",
+    "dask-cudf",
+    "cuspatial",
+    "cuproj",
+    "cuml",
+    "cugraph",
+    "cudf",
+    "ptxcompiler",
+    "cubinlinker",
+    "cugraph-dgl",
+    "cugraph-pyg",
+    "cugraph-equivariant",
+    "raft-dask",
+    "pylibwholegraph",
+    "pylibraft",
+    "cuxfilter",
+    "cucim",
+    "ucx-py",
+    "ucxx",
+    "pynvjitlink",
+    "distributed-ucxx",
+}
+
+RAPIDS_CUDA_SUFFIXED_PACKAGES = set(RAPIDS_ALPHA_SPEC_PACKAGES)
+
+ALPHA_SPECIFIER = ">=0.0.0a0"
+
+ALPHA_SPEC_OUTPUT_TYPES = {
+    "pyproject",
+    "requirements",
+}
+
+
+def node_has_type(node, tag_type):
+    return node.tag == f"tag:yaml.org,2002:{tag_type}"
+
+
+def is_rapids_cuda_suffixed_package(name):
+    return any(
+        name.startswith(f"{package}-cu") for package in RAPIDS_CUDA_SUFFIXED_PACKAGES
```

Comment:
This would only be a problem if `dask` was in `RAPIDS_CUDA_SUFFIXED_PACKAGES`. `dask-cuda` being in the list isn't a problem, because it will just search for `dask-cuda-cu`.